symfony/plugins/orangehrmAdminPlugin/Dao/LanguageDao.php

```diff
@@ -0,0 +1,171 @@
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Admin\Dao;
+
+use Exception;
+use OrangeHRM\Admin\Dto\LanguageSearchFilterParams;
+use OrangeHRM\Core\Dao\BaseDao;
+use OrangeHRM\Core\Exception\DaoException;
+use OrangeHRM\Entity\Language;
+use OrangeHRM\ORM\Paginator;
+
+class LanguageDao extends BaseDao
+{
+
+    /**
+     * @param Language $language
+     * @return Language
+     * @throws DaoException
+     */
+    public function saveLanguage(Language $language): Language
+    {
+        try {
+            $this->persist($language);
+            return $language;
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param int $id
+     * @return Language|null
+     * @throws DaoException
+     */
+    public function getLanguageById(int $id): ?Language
+    {
+        try {
+            $language = $this->getRepository(Language::class)->find($id);
+            if ($language instanceof Language) {
+                return $language;
+            }
+            return null;
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param string $name
+     * @return Language|null
+     * @throws DaoException
+     */
+    public function getLanguageByName(string $name): ?Language
+    {
+        try {
+            $query = $this->createQueryBuilder(Language::class, 'l');
+            $trimmed = trim($name, ' ');
+            $query->andWhere('l.name = :name');
+            $query->setParameter('name', $trimmed);
+            return $query->getQuery()->getOneOrNullResult();
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param LanguageSearchFilterParams $languageSearchFilterParams
+     * @return array
+     * @throws DaoException
+     */
+    public function getLanguageList(LanguageSearchFilterParams $languageSearchFilterParams): array
+    {
+        try {
+            $paginator = $this->getLanguageListPaginator($languageSearchFilterParams);
+            return $paginator->getQuery()->execute();
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param LanguageSearchFilterParams $languageSearchFilterParams
+     * @return Paginator
+     */
+    public function getLanguageListPaginator(LanguageSearchFilterParams $languageSearchFilterParams
+    ): Paginator {
+        $q = $this->createQueryBuilder(Language::class, 'l');
+        if (!is_null($languageSearchFilterParams->getSortField())) {
```

Comment:
remove 
```php
        if (!is_null($languageSearchFilterParams->getSortField())) {
            $q->addOrderBy($languageSearchFilterParams->getSortField(), $languageSearchFilterParams->getSortOrder());
        }
        if (!empty($languageSearchFilterParams->getLimit())) {
            $q->setFirstResult($languageSearchFilterParams->getOffset())
                ->setMaxResults($languageSearchFilterParams->getLimit());
        }
```
and follow this pattern
```php
$this->setSortingAndPaginationParams($q, $languageSearchFilterParams);
```
https://github.com/orangehrm/orangehrm/blob/5.0/symfony/plugins/orangehrmPimPlugin/Dao/EmployeeDao.php#L68